merino/providers/amo/backends/dynamic.py

```diff
@@ -55,7 +62,9 @@ async def _fetch_addon(
                 "number_of_ratings": json_res["ratings"]["count"],
             }
         except httpx.HTTPError as e:
-            logger.error(f"Addons API could not find key: {addon_key}, {e}")
+            logger.error(
+                f"Addons API could not find key: {addon_key}, {e}, {e.__class__}"
+            )
```

Comment:
Same as above. 
`HTTPStatusError("Client error '400 Bad Request' for url 'https://addons.mozilla.org/api/v5/addons/addon/video-downloadhelper'\nFor more information check: https://httpstatuses.com/400")` for `e`

`<class 'httpx.HTTPStatusError'>` for `e.__class__`


```suggestion
            logger.error(
                f"Addons API could not find key: {addon_key}, {e}"
            )
```

Reply:
Same as above. This is mainly logging a sanity check, as this _should_ be `HttpStatusError`. It would be interesting to see if this catches any other types of errors as well.